web/src/containers/ApdPreviousActivityTableMMIS.js

```diff
@@ -0,0 +1,144 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { formatMoney } from '../util/formats';
+import { DollarInput } from '../components/Inputs';
+import { updateApd } from '../actions/apd';
+import { t } from '../i18n';
+
+const i18nBase = { scope: 'previousActivities.actualExpenses.table' };
+
+const borderClass = i => `border-left border-${i < 0 ? 'gray' : 'aqua'}`;
+
+const rollup = previous => {
+  const total = {
+    actual: previous.mmis.federalActual + previous.mmis.stateActual,
+    approved: previous.mmis.federalApproved + previous.mmis.stateApproved
+  };
+  return total;
+};
+
+const ApdPreviousActivityTable = ({
+  previousActivityExpenses,
+  updateApd: dispatchUpdateApd
+}) => {
+  const years = Object.keys(previousActivityExpenses);
```

Comment:
one area for future refinement: moving this to `mapStateToProps` and incorporating the `const total = rollup(previousActivityExpenses[year])` logic there so that the render method gets simpler